jakttest/jakttest.jakt

```diff
@@ -10,48 +10,57 @@ import parser { Parser }
 import lexer { Lexer }
 import utility { panic }
 
-namespace fs {
+// needed for sigaction() stuff
+import extern c "signal.h" {}
 
-    // subset of `struct stat` from <sys/stat.h> that contains
-    // just what is used.
-    extern struct StatResults {
-        function modified_time(this) -> usize
-        // Whether the file is executable by the current user
-        function is_executable(this) -> bool
-        function is_regular_file(this) -> bool
-        function is_directory(this) -> bool
-    }
+import extern "fs.h" {
+    namespace fs {
+        // subset of `struct stat` from <sys/stat.h> that contains
+        // just what is used.
+        extern struct StatResults {
+            function modified_time(this) -> usize
+            // Whether the file is executable by the current user
+            function is_executable(this) -> bool
+            function is_regular_file(this) -> bool
+            function is_directory(this) -> bool
+        }
 
-    extern class DirectoryIterator {
-        // Get next listing name. Note that the full path is not given,
-        // just the base name.
-        public function next(mut this) throws -> String?
-    }
-    // Get a listing of a directory, using readdir().
-    extern function list_directory(path: String) throws -> DirectoryIterator
-    // Returns whether the given path is a directory
+        extern class DirectoryIterator {
+            // Get next listing name. Note that the full path is not given,
+            // just the base name.
+            public function next(mut this) throws -> String?
+        }
+        // Get a listing of a directory, using readdir().
+        extern function list_directory(path: String) throws -> DirectoryIterator
+        // Returns whether the given path is a directory
+
+        // Create a directory
+        extern function mkdir(path: String) throws
+        
+        // Remove a directory
+        extern function rmdir(path: String) throws
 
-    // Create a directory
-    extern function mkdir(path: String) throws
-    
-    // Remove a directory
-    extern function rmdir(path: String) throws
+        // Remove a file
+        extern function unlink(path: String) throws
 
-    // Remove a file
-    extern function unlink(path: String) throws
+        // Calls `stat` on a file. Silences `ENOENT` error by setting
+        // StatResults.exists to false.
+        extern function stat_silencing_enoent(path: String) throws -> StatResults?
 
-    // Calls `stat` on a file. Silences `ENOENT` error by setting
-    // StatResults.exists to false.
-    extern function stat_silencing_enoent(path: String) throws -> StatResults?
 
+    }
+}
 
+// FIXME: introducing this inside a namespace seems to override the namespace
+// from import extern
+//namespace fs {
```

Comment:
remove

Reply:
why? It has a reason to be commented. But maybe just putting in the fixme that it should be inside a namespace `fs` is enough. Thanks